File: Daily_Dsa/Java/Array/ReverseAnArray.java
import java.util.Scanner;
class ReverseAnArray{
    public static void swap(int[]arr,int a,int b){
        int temp=arr[a];
        arr[a]=arr[b];
        arr[b]=temp;
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        String inputString =sc.nextLine();
        System.out.println(inputString);
        String[] strarray=inputString.split(" ");
        int[] inputArray= new int[strarray.length];
        for (int i=0;i<strarray.length;i++){
            
            inputArray[i]=Integer.parseInt(strarray[i]);
        }
        int[] arr=inputArray;
        int left=0;
        int right=arr.length-1;
        while(left<=right){
            swap(arr, left, right);
            left++;
            right--;
        }
        for(int a:arr){
            System.out.println(a);
        }
    }
}

File: Daily_Dsa/Python/Array/ReverseAnArray.py
def swap(arr, a, b):
    arr[a], arr[b] = arr[b], arr[a]


arr = list(map(int, input().strip().split()))

left, right = 0, len(arr) - 1
while left < right:  # Optimized condition
    swap(arr, left, right)
    left += 1
    right -= 1

for num in arr:
    print(num)
